src/sparseml/pytorch/optim/modifier_quantization.py

```diff
@@ -428,6 +468,61 @@ def _enable_module_qat(self, module: Module):
         if self._quantize_embeddings:
             prepare_embeddings_qat(module, reduce_range=self._reduce_range)
         self._qat_enabled = True
+        self._calibrate_if_possible(module)
+
+    def _calibrate_if_possible(self, module):
+        if self.num_calibration_steps == 0 and self._calibration_dataloader:
+            warnings.warn(
+                f"num_calibration_steps is {self.num_calibration_steps}."
+                f"Calibration data loader will not be used."
+            )
+        elif self.num_calibration_steps and not self._calibration_dataloader:
+            raise ValueError(
+                f"num_calibration_steps is {self.num_calibration_steps}. "
+                "Calibration data loader is not set."
```

Comment:
let's specify that it should be passed in as `calibration_dataloader` on initialize

Reply:
Done!